pkg/agent/handler/apiservicehandler.go

```diff
@@ -0,0 +1,44 @@
+package handler
+
+import (
+	"fmt"
+
+	agentcache "github.com/Axway/agent-sdk/pkg/agent/cache"
+	v1 "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/api/v1"
+	"github.com/Axway/agent-sdk/pkg/apic/definitions"
+	"github.com/Axway/agent-sdk/pkg/watchmanager/proto"
+)
+
+const apiService = "APIService"
+
+type apiSvcHandler struct {
+	agentCacheManager agentcache.Manager
+}
+
+// NewAPISvcHandler creates a Handler for API Services.
+func NewAPISvcHandler(agentCacheManager agentcache.Manager) Handler {
+	return &apiSvcHandler{
+		agentCacheManager: agentCacheManager,
+	}
+}
+
+func (h *apiSvcHandler) Handle(action proto.Event_Type, _ *proto.EventMeta, resource *v1.ResourceInstance) error {
+	if resource.Kind != apiService {
+		return nil
+	}
+
+	id, ok := resource.Attributes[definitions.AttrExternalAPIID]
```

Comment:
Keep in mind this will need to be updated to read from the sub resource.

Reply:
Sure thing, I just broke this out from the single handlers file, will merge those changes when they are in main

Reply:
I updated it on the migration branch, which will probably be merged before this so it should be fine.